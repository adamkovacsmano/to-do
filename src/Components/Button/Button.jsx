import React, { Component } from "react";
import styles from "./Button.module.scss";

class Button extends Component {
  render() {
    return (
      <button
        id="testButton2"
        className={styles.primary}
        onClick={this.props.onClick}
      >
        {this.props.name}
      </button>
    );
  }
}

export default Button;
