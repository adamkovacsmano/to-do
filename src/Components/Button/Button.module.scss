.primary {
  background-color: transparent;
  color: rgb(44, 84, 148);
  border: none;
  text-decoration: underline;
  cursor: pointer;
  font-weight: bolder;
}
