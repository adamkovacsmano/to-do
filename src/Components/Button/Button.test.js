import React from "react";
import Button from "./Button";
import { shallow, mount } from "enzyme";

describe("Button Tests", () => {
  let button;
  beforeEach(() => {
    button = shallow(<Button />);
  });

  test("should match the snapshot", () => {
    expect(button).toMatchSnapshot();
  });
});

const clickFn = jest.fn();
describe("button", () => {
  it("button click should filter component", () => {
    const component = shallow(<Button onClick={clickFn} />);
    component.find("button#testButton2").simulate("click");
    expect(clickFn).toHaveBeenCalled();
  });
});

it("should be possible to activate Button with onClick", () => {
  const component = mount(<Button />);
  component.find("button#testButton2").simulate("click");
  expect(component).toMatchSnapshot();
  component.unmount();
});
