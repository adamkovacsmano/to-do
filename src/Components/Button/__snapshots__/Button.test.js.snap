// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests should match the snapshot 1`] = `
<button
  className="primary"
  id="testButton2"
/>
`;

exports[`FilterIcon Tests should match the snapshot 1`] = `
<button
  className="primary"
  id="testButton2"
/>
`;

exports[`should be possible to activate Button with onClick 1`] = `
<Button>
  <button
    className="primary"
    id="testButton2"
  />
</Button>
`;
